apps/site/assets/ts/models/alert.ts

```diff
@@ -19,6 +20,93 @@ export const alertsByStop = (alerts: Alert[], stopId: StopId): Alert[] =>
       !!entities.stop && entities.stop!.some((id: StopId) => id === stopId)
   );
 
+const hasEffect = (alerts: Alert[], effect: string): boolean =>
+  some(alerts, alert => alert.effect === effect);
+
+export const isSuspension = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "suspension");
+
+export const isShuttleService = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "shuttle");
+
+export const isDetour = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "detour");
+
+export const alertsWithStop = (alerts: Alert[]): Alert[] =>
+  alerts.filter(
+    ({ informed_entity: entites }: Alert): boolean => !!entites.stop
+  );
+
+export const routeWideAlerts = (alerts: Alert[]): Alert[] =>
+  alerts.filter(({ informed_entity: { entities } }: Alert): boolean =>
+    entities.some(entity => !entity.stop && !entity.trip)
+  );
+
+// Can there be alerts that don't affect routes
+export const alertsByRoute = (alerts: Alert[]): { [key: string]: Alert[] } => {
+  return reduce(
+    alerts,
+    (acc, alert) => {
+      // create a map of every route id to the alert
+      const explodedAlerts = reduce(
```

Comment:
suggestion (non-blocking): `exploded` to me doesn't imply what the comment suggests this data type is. Maybe `alertByRoute` would be helpful?

Also non-blocking: consider using lodash [group_by](https://lodash.com/docs/4.17.15#groupBy) to simplify this function

Reply:
Ah thanks, I missed that the alerts => multiple routes aspect. It looks to me like on line 54 a routeId is only mapped to a list containing a single alert.

Maybe [flatMap](https://lodash.com/docs/4.17.15#flatMap) could be used to get a flattened list of [{route_id, alert}], and `groupBy` could be called on that flattened list?

Reply:
@kotva006 is there an existing test case for when there is more than one route associated with an alert? Want to confirm this is working as expected

Reply:
@KaylaBrady yeah [this test](https://github.com/mbta/dotcom/blob/61d64317396a890dcaf344d1fc33f46983ca1b12/apps/site/assets/ts/models/__tests__/alert-test.ts#L255) tests that a single route can have multiple alerts

Reply:
Thank you! I see the problem now with how I was interpreting this function.

Now I understand: `alert.informed_entity.route` is a list of unique route ids for an alert, so each route/alert combo added to the accumulator on line 53 is unique. 

Still non-blocking feedback, but perhaps some of the suggested functions above help simplify this function so it is easier to read. Alternatively, consider giving the accumulators more descriptive names or breaking out the nested reduces into helper functions.